src/compiler/checker.ts

```diff
@@ -40298,7 +40381,9 @@ namespace ts {
         }
 
         function getNodeCheckFlags(node: Node): NodeCheckFlags {
-            return getNodeLinks(node).flags || 0;
+            const nodeId = node.id || 0;
+            if (nodeId < 0 || nodeId >= nodeLinks.length) return 0;

```

Comment:
what could cause an nodeId that's not in nodeLinks? Is nodeLinks only filled in for nodes with NodeCheckFlags set?

Reply:
There are a number of places where transformers create node ids that may not be present in `NodeLinks`. The code this replaces would unconditionally add an entry to `nodeLinks` for nodes that may not have a value.